src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -0,0 +1,204 @@
+//-----------------------------------------------------------------------
+// <copyright file="SolutionBindingOperation.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using EnvDTE;
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    /// <summary>
+    /// Solution level binding by delegating some of the work to <see cref="ProjectBindingOperation"/>
+    /// </summary>
+    internal class SolutionBindingOperation : IBindingOperation, ISolutionRuleStore

```

Comment:
Why does `SolutionBindingOperation` need to implement `ISolutionRuleStore`?
The only use of the interface within this class is to pass `this` into the `new ProjectBindingOperation` ctor.
Why can't SolutionRuleStore be it's own class implementing the interface, and have it be passed-into/constructed by this class? That would also mean that in the `BindingWorkflow` ctor, there wouldn't be two fields which point to the same object instance (but with a different type).